File: selection-sort/JavaScript/selection-sort.js

function selectionSort(arr){
    for(let i = 0; i < arr.length; i++){
        let currentMinIndex = i;
        for(let j = currentMinIndex+1; j < arr.length; j++){
            if(arr[j] < arr[currentMinIndex]){
                currentMinIndex = j;
            }
        }

        if(currentMinIndex != i){
            let originalMinValue = arr[i];
            arr[i] = arr[currentMinIndex];
            arr[currentMinIndex] = originalMinValue;
        }
    }

    return arr;
}

let unsorted = [12,2,7,4,5,6,3,8,1,0,9,10,11];
let sorted = selectionSort(unsorted);
console.log(sorted);

File: selection-sort/Python/selectionsort.py

def selection_sort(items):
    for i in range(len(items)):
        currentMinIndex = i
        for j in range(i + 1, len(items)):
            if items[j] < items[currentMinIndex]:
                currentMinIndex = j
        
        if currentMinIndex != i:
            originalMinValue = items[i]
            items[i] = items[currentMinIndex]
            items[currentMinIndex] = originalMinValue
        
    return items


unsorted = [12,2,7,4,5,6,3,8,1,0,9,10,11]
sorted = selection_sort(unsorted)
print(sorted)